src/main/java/edu/ksu/canvas/attendance/services/SynchronizationService.java

```diff
@@ -114,29 +114,27 @@ private AttendanceCourse synchronizeCourseFromCanvasToDb(long canvasCourseId) {
 
     private List<AttendanceStudent> synchronizeStudentsFromCanvasToDb(Map<Section, List<Enrollment>> canvasSectionMap, boolean hasOneAuthorityRole) {
         List<AttendanceStudent> ret = new ArrayList<>();
-        List<AttendanceStudent> existingStudentsInDb = null;
-        Set<AttendanceStudent> droppedStudents = new HashSet<>();
 
         for(Section section: canvasSectionMap.keySet()) {
-
-            if(existingStudentsInDb == null) {
-                existingStudentsInDb = studentRepository.findByCanvasCourseId(section.getCourseId());
-                droppedStudents.addAll(existingStudentsInDb);
-            }
+            Set<AttendanceStudent> droppedStudents = new HashSet<>();
+            List<AttendanceStudent> existingStudentsInDb;
+            existingStudentsInDb = studentRepository.findByCanvasSectionIdOrderByNameAsc(section.getId());
+            droppedStudents.addAll(existingStudentsInDb);
 
             for(Enrollment enrollment: canvasSectionMap.get(section)) {
 
                 Optional<AttendanceStudent> foundUser =
                         existingStudentsInDb.stream()
                                         .filter(u -> u.getSisUserId().equals(enrollment.getUser().getSisUserId()))
+                                        .limit(1)
                                         .findFirst();
 
                 if (foundUser.isPresent()){
                     droppedStudents.remove(foundUser.get());
+                    foundUser.get().setDeleted(Boolean.FALSE);
                 }
                 AttendanceStudent student = foundUser.isPresent() ? foundUser.get() : new AttendanceStudent();
```

Comment:
this isn't something you've changed but this line could maybe be simplified by making use of the optional method orElse.. like foundUser.orElse(new AttendanceStudent()); maybe